nucliadb_utils/src/nucliadb_utils/storages/storage.py

```diff
@@ -532,6 +525,19 @@ async def get_stream_message(self, key: str) -> bytes:
     async def del_stream_message(self, key: str) -> None:
         await self.delete_upload(key, cast(str, self.indexing_bucket))
 
+    @abc.abstractmethod
+    async def insert_object(self, bucket: str, key: str, data: bytes) -> None:
+        """
+        Put some binary data into the object storage without any object metadata.
+        """
+        ...
+
+    async def upload(self, bucket: str, key: str, data: bytes) -> None:
```

Comment:
I feel that you're introducing a new confusing concept here. The difference between inserting or uploading is unknown except if you read the implementation.
Could you go for something more explicit, knowing that `uploadbytes` is a multipart upload and `insert_object` a upload in one go?